compiler/rustc_mir/src/monomorphize/collector.rs

```diff
@@ -59,11 +59,14 @@
 //!
 //! ### Discovering roots
 //!
-//! The roots of the mono item graph correspond to the non-generic
+//! The roots of the mono item graph correspond to the public non-generic
 //! syntactic items in the source code. We find them by walking the HIR of the
-//! crate, and whenever we hit upon a function, method, or static item, we
-//! create a mono item consisting of the items DefId and, since we only
-//! consider non-generic items, an empty type-substitution set.
+//! crate, and whenever we hit upon a public function, method, or static item,
+//! we create a mono item consisting of the items DefId and, since we only
+//! consider non-generic items, an empty type-substitution set. (In eager
+//! collection mode, during incremental compilation, all non-generic functions
+//! are considered as roots, as well as when the `-Clink-dead-code` option is
+//! specified. Functions marked `#[no_mangle]` also always act as roots.)
```

Comment:
You are missing functions called by inlinable functions.

Reply:
Are you sure it's "functions called by inlinable functions" and not just "inlinable functions" that are roots? I am not sure what part of the code is responsible for that.

Reply:
Yes, the `reachable_set` includes all functions called by inlinable functions. https://github.com/rust-lang/rust/blob/673d0db5e393e9c64897005b470bfeb6d5aec61b/compiler/rustc_passes/src/reachable.rs#L106-L108